assets/pages/digital-subscription-checkout/components/checkoutForm.jsx

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import { countries } from 'helpers/internationalisation/country';
+
+import { Input } from 'components/forms/standardFields/input';
+import { Select } from 'components/forms/standardFields/select';
+import { withLabel } from 'components/forms/formHOCs/withLabel';
+import { asControlled } from 'components/forms/formHOCs/asControlled';
+
+import { type State, actions } from '../digitalSubscriptionCheckoutReducer';
+
+
+// ----- Form Fields ----- //
+
+function Input1(props: { id: string, label: string, value: string, setValue: string => void }) {
+  const Com = compose(asControlled, withLabel)(Input);
+  return <Com {...props} />;
+}
+
+function Select1(props: { id: string, label: string, value: string, setValue: string => void }) {
+  const Com = compose(asControlled, withLabel)(Select);
+  return <Com {...props} />;
+}
+
+const FirstName = connect(
+  (state: State) => ({ value: state.page.firstName }),
+  { setValue: actions.setFirstName },
+)(Input1);
+
+const LastName = connect(
+  (state: State) => ({ value: state.page.lastName }),
+  { setValue: actions.setLastName },
+)(Input1);
+
+const Telephone = connect(
+  (state: State) => ({ value: state.page.telephone }),
+  { setValue: actions.setTelephone },
+)(Input1);
+
+const Country = connect(
+  (state: State) => ({ value: state.page.country }),
+  { setValue: actions.setCountry },
+)(Select1);
+
+
+// ----- Component ----- //
+
+function CheckoutForm() {
+
+  return (
+    <div>
+      <FirstName id="first-name" label="First name" type="text" />
```

Comment:
A slightly nicer API would be for these fields to know their type. That is, `FirstName` knows its `text` and `Telephone` a `tel` as I the `type` wouldn't change.

Reply:
True. I could do that by creating another intermediate component, partially applying the prop. However, wouldn't the id and label also be specific to this field in that case too?

Reply:
Yes I think you could very easily just subsume the id and label into the component